cluster/slave/slave_grpc.go

```diff
@@ -737,14 +737,23 @@ func (s *SlaveServerSideOp) AddTransactions(ctx context.Context, req *rpc.Reques
 
 func (s *SlaveServerSideOp) HandleNewMinorBlock(ctx context.Context, req *rpc.Request) (*rpc.Response, error) {
 	var (
-		gReq     types.MinorBlock
+		gReq     rpc.P2PRedirectRequest
+		mblock   p2p.NewBlockMinor
 		response = &rpc.Response{RpcId: req.RpcId}
 		err      error
 	)
 	if err = serialize.DeserializeFromBytes(req.Data, &gReq); err != nil {
 		return nil, err
 	}
-	if err = s.slave.NewMinorBlock(&gReq); err != nil {
+	if err = serialize.DeserializeFromBytes(gReq.Data, &mblock); err != nil {
+		return nil, err
+	}
+	if gReq.Branch != mblock.Block.Branch().Value {
+		return nil, fmt.Errorf("invalid NewBlockMinor Request: mismatch branch value from peer %v. in request meta: %d, in minor header: %d",
+			gReq.PeerID, gReq.Branch, mblock.Block.Branch().Value)
+
+	}
+	if err = s.slave.NewMinorBlock(mblock.Block); err != nil {
```

Comment:
add peerid for broadcast or use p2p.NewBlockMinor instead of rpc.P2PRedirectRequest